core/src/test/resources/json/shaped_recipe.json

```diff
@@ -0,0 +1,23 @@
+{
+  "key" : {
+    "namespace" : "minecraft",
+    "key" : "test"
+  },
+  "result" : {
+    "material" : "STONE",
+    "amount" : 32
+  },
+  "shape" : [ "aaa", " b ", " b " ],
+  "group" : "",
+  "ingredients" : {
+    " " : null,
```

Comment:
What happens, if you don't set the Strings.EMPTY ingredient to null?

Reply:
Thats the neat part you dont. This is just an empty map entry, which is also actually not allowed by spigot in recipes, but somehow still possible. I will add an test for this.